publication-model/src/main/java/no/unit/nva/model/associatedartifacts/file/FileResponse.java

```diff
@@ -0,0 +1,158 @@
+    @JsonProperty(FileResponse.ALLOWED_OPERATIONS_FIELD) Set<FileOperation> allowedOperations
+) implements AssociatedArtifactResponse {
+
+    public static final String ALLOWED_OPERATIONS_FIELD = "allowedOperations";
+    public static final String TYPE_NAME_FIELD = "type";
+
+    @Override
+    public String getArtifactType() {
+        return type;
+    }
+
+    // Static Builder class
+    public static class Builder {
+        private UUID identifier;
+        private String type;
+        private String name;
+        private String mimeType;
+        private Long size;
+        private URI license;
+        private PublisherVersion publisherVersion;
+        private Instant embargoDate;
+        private RightsRetentionStrategy rightsRetentionStrategy;
+        private String legalNote;
+        private Instant publishedDate;
+        private UploadDetails uploadDetails;
+        private Set<FileOperation> allowedOperations;
+
+        // Builder methods to set each field
+        public Builder withIdentifier(UUID identifier) {
+            this.identifier = identifier;
+            return this;
+        }
+
+        public Builder withType(String type) {
+            this.type = type;
+            return this;
+        }
+
+        public Builder withName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        public Builder withMimeType(String mimeType) {
+            this.mimeType = mimeType;
+            return this;
+        }
+
+        public Builder withSize(Long size) {
+            this.size = size;
+            return this;
+        }
+
+        public Builder withLicense(URI license) {
+            this.license = license;
+            return this;
+        }
+
+        public Builder withPublisherVersion(PublisherVersion publisherVersion) {
+            this.publisherVersion = publisherVersion;
+            return this;
+        }
+
+        public Builder withEmbargoDate(Instant embargoDate) {
+            this.embargoDate = embargoDate;
+            return this;
+        }
+
+        public Builder withRightsRetentionStrategy(RightsRetentionStrategy rightsRetentionStrategy) {
+            this.rightsRetentionStrategy = rightsRetentionStrategy;
+            return this;
+        }
+
+        public Builder withLegalNote(String legalNote) {
+            this.legalNote = legalNote;
+            return this;
+        }
+
+        public Builder withPublishedDate(Instant publishedDate) {
+            this.publishedDate = publishedDate;
+            return this;
+        }
+
+        public Builder withUploadDetails(UploadDetails uploadDetails) {
+            this.uploadDetails = uploadDetails;
+            return this;
+        }
+
+        public Builder withAllowedOperations(Set<FileOperation> allowedOperations) {
+            this.allowedOperations = allowedOperations;
+            return this;
+        }
+
+        // Build method to create a new FileResponse instance
```

Comment:
no need for comment?

Reply:
Any of them. They state the obvious from the code.

Reply:
so busted using gpt 😂